src/Place/ReadModel/History/HistoryProjector.php

```diff
@@ -137,12 +135,6 @@ public function handle(DomainMessage $domainMessage): void
             case $event instanceof MajorInfoUpdated:
                 $this->projectMajorInfoUpdated($domainMessage);
                 break;
-            case $event instanceof MarkedAsCanonical:
-                $this->projectMarkedAsCanonical($domainMessage);
-                break;
```

Comment:
Open question(since I'm not really familiar with this part of the code). But since the eventstore already has MarkAsDuplicate and MarkAsCanonical Events, would it be safe to remove them here?

Reply:
It is safe to remove the handling of `MarkedAsDuplicate` and `MarkedAsCanonical` in every event listener. It is even safe to remove the `apply` method from an aggregate. The only thing that is still required to be around are the events.

You can check that on `EventSourcedAggregateRoot::handle`. If the method does not exist, it is not handled.